### ISSUE TITLE
Infinite loop in widget.js when used in an inline with an empty/zero-length queryset.

### ISSUE THREAD

**Issue by darylyu** [jazzband/django-sortedm2m#97]
This particular loop goes on forever: https://github.com/gregmuellegger/django-sortedm2m/blob/master/sortedm2m/static/sortedm2m/widget.js#L22

```
while (!label || !label.length) {
  currentElement = currentElement.parent();
  label = currentElement.siblings('label');
}
```

It only happens if the item that needs sorting is empty and uses a Django admin inline. It works fine if it's not a Django admin inline - when viewing it from that model's page on the Django admin.

I've added logging inside the loop:
```
while (!label || !label.length) {
  console.log("here");
  console.log(label);
  console.log(currentElement);
  currentElement = currentElement.parent();
  label = currentElement.siblings('label');
}
```

This is the output:
```
here
undefined
[ul.sortedm2m-items.hide.sortedm2m, context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div.sortedm2m-container, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div.related-widget-wrapper, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[td.field-deals, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[tr#deal_categories-0.form-row.row1.has_original.dynamic-deal_categories, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[tbody, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[table, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[fieldset.module, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div.tabular.inline-related.last-related, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div#deal_categories-group.inline-group, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[form#hotel_form, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div#content-main, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div#content.colM, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[div#container, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[body.flat-theme.app-hotels.model-hotel.change-form, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[html, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
[document, prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[1], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
```

It then goes on to print this forever:
```
here
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
[prevObject: jQuery.fn.init[0], context: ul.sortedm2m-items.hide.sortedm2m]
```

**Comment by gregmuellegger** [comment]
Thanks for the report! That's unfortunate to have this bug in the package. I myself don't use the admin widget so I haven't stumbled upon this problem.

Are you interested to work on a fix and provide a pull request? That would ensure that the issue gets resolved.

**Comment by darylyu** [comment]
@gregmuellegger I'm not familiar with the code, but I'm willing to give it a try if you'll provide guidance. My worry is I really won't be able to tell if I break it for other people.

Where do I start? My idea is to massage the data so the data that gets passed to widget.js is the same as the data that gets passed to it if the widget isn't an admin inline.

**Comment by hradhak** [comment]
I've encountered this same issue.  Any progress on a fix @darylyu  ?

Thanks

**Comment by netral23** [comment]
Just check ul element on empty in iterateUl . Works fine for me.
```
function iterateUl() {
            $('.sortedm2m-items').each(function () {
                var ul = $(this);
                if($.trim(ul.html()))
                    prepareUl(ul);
                ul.removeClass('hide');
            });
}
```

**Comment by nemesisdesign** [comment]
Was this fixed by https://github.com/jazzband/django-sortedm2m/commit/b8e8326a69b2dd9942ed0a21acb903268b34d0b2?

@clintonb

**Comment by clintonb** [comment]
Fixed by #170
